sssd_test_framework/utils/authentication.py

```diff
@@ -320,22 +320,30 @@ def passkey(self, username: str, *, pin: str | int, device: str, ioctl: str, scr
         :return: True if authentication was successful, False otherwise.
         :rtype: bool
         """
-        self.fs.backup("/etc/sysconfig/sssd")
+        self.fs.backup("/usr/libexec/sssd/passkey_child")
+        self.fs.copy("/usr/libexec/sssd/passkey_child", "/usr/libexec/sssd/passkey_child.orig")
+        self.host.ssh.run("rm -f /usr/libexec/sssd/passkey_child", raise_on_error=False)
+
         device_path = self.fs.upload_to_tmp(device, mode="a=r")
         ioctl_path = self.fs.upload_to_tmp(ioctl, mode="a=r")
         script_path = self.fs.upload_to_tmp(script, mode="a=r")
 
         run_su = self.fs.mktmp(
             rf"""
-            #!/bin/bash
-            set -ex
-            env | grep ^UMOCKDEV_ > /etc/sysconfig/sssd
-            printf "LD_PRELOAD=$LD_PRELOAD" >> /etc/sysconfig/sssd
-            systemctl restart sssd
-            chmod -R a+rwx $UMOCKDEV_DIR
-
-            su --shell /bin/sh nobody -c "su - '{username}'"
-            """,
+                #!/bin/bash
+                set -ex
+                echo '#!/bin/bash' > /usr/libexec/sssd/passkey_child
+                echo -n 'export ' >> /usr/libexec/sssd/passkey_child
+                env | grep ^UMOCKDEV_ >> /usr/libexec/sssd/passkey_child
+                echo -n 'export ' >> /usr/libexec/sssd/passkey_child
+                printf "LD_PRELOAD=$LD_PRELOAD\n" >> /usr/libexec/sssd/passkey_child
+                echo 'exec /usr/libexec/sssd/passkey_child.orig $@' >> /usr/libexec/sssd/passkey_child
+                chmod 755 /usr/libexec/sssd/passkey_child
+                systemctl restart sssd
+                chmod -R a+rwx $UMOCKDEV_DIR
```

Comment:
What's in UMOCKDEV_DIR? Is this really needed?

Reply:
UMOCKDEV_DIR is a temporary working directory where it does some of it's emulation.   It was needed with the original solution with su to make it work as expected so I expect it is needed here as well.

Reply:
The UMOCKDEV_DIR name is generated randomly every time umockdev is run. Well, maybe there's some logic behind it but to me it looks completely random.

Reply:
yes, we need it and every time it create the random directory under /tmp that starts with` umockdev` prefix
like
export UMOCKDEV_DIR=/tmp/umockdev.8LORD2
export UMOCKDEV_DIR=/tmp/umockdev.DVC2D2